### PR TITLE
Add icon support for quickpick custom button

### PR DESCRIPTION
**Pull request by andreamah**
image

Related to #201735

Added support for icon in quickpick custom button and adopted it in quick search.

### CODE REVIEW

File: src/vs/platform/quickinput/common/quickInput.ts [hunk 1]
```diff
@@ -404,6 +405,11 @@ export interface IQuickPick<T extends IQuickPickItem> extends IQuickInput {
 	 */
 	customLabel: string | undefined;
 
+	/**
+	 * The icon for the custom button. Note: do not set customLabel if you set customIcon, otherwise the label formatting will be wrong.
```
Comment (andreamah) [inline src/vs/platform/quickinput/common/quickInput.ts:409]: Not sure why this is or how to fix it, but formatting was wrong when this happened. However, the `Button` object allows us to set both label and icon, so I replicated that behavior here.

File: src/vs/platform/quickinput/browser/quickInput.ts [hunk 2]
```diff
@@ -530,6 +530,7 @@ export class QuickPick<T extends IQuickPickItem> extends QuickInput implements I
 	private _ok: boolean | 'default' = 'default';
 	private _customButton = false;
 	private _customButtonLabel: string | undefined;
+	private _customButtonIcon: ThemeIcon | undefined;
```
Comment (TylerLeonhardt) [inline src/vs/platform/quickinput/browser/quickInput.ts:533]: I'd rather you keep using label, but have it be `$(icon-name)`

Comment (andreamah) [reply src/vs/platform/quickinput/browser/quickInput.ts:533]: I think that the button ends up having a little more padding on each side, but I think that's fine.
image

File: src/vs/platform/quickinput/browser/quickInput.ts [hunk 3]
```diff
@@ -745,20 +745,11 @@ export class QuickPick<T extends IQuickPickItem> extends QuickInput implements I
 		return this._customButtonLabel;
 	}
 
-	set customLabel(label: string | undefined) {
+	set customLabel(label: string | IMarkdownString | undefined) {
```
Comment (TylerLeonhardt) [inline src/vs/platform/quickinput/browser/quickInput.ts:748]: can you try using this instead:
https://github.com/microsoft/vscode/blob/44ffc28fea5da56f1e447939da3621d644fe86ce/src/vs/base/browser/ui/button/button.ts#L26

then the API doesn't have to change at all

**Review by fellotin — APPROVED** [review]

**Review by fellotin — APPROVED** [review]

**Review by 0ME9A — APPROVED** [review]

**Review by TylerLeonhardt — APPROVED** [review]

**Pull request merged**